### PR TITLE
FSET-1658. Add candidate to FSB event

### PR DESCRIPTION
**Pull request by vglushak**

### CODE REVIEW

File: app/repositories/application/GeneralApplicationRepository.scala [hunk 1]
```diff
@@ -850,14 +852,29 @@ class GeneralApplicationMongoRepository(
...
+    schemeId: Option[SchemeId]
+  ): Future[CandidatesEligibleForEventResponse] = {
+    val appStatus = eventType.applicationStatus
+
+    val status = EventProgressStatuses.get(appStatus)
+    val awaitingAllocation = status.awaitingAllocation.key
+    val confirmedAllocation = status.allocationConfirmed.key
+    val unconfirmedAllocation = status.allocationUnconfirmed.key
+
+    // TODO: do we need location based check for FSB ? What about Virtual.
```
Comment (kjharkin) [inline app/repositories/application/GeneralApplicationRepository.scala]: remove comment

File: it/model/report/ProgressStatusesMongoReportLabelsSpec.scala [hunk 2]
```diff
@@ -34,7 +34,8 @@ class ProgressStatusesMongoReportLabelsSpec extends MongoRepositorySpec {
   "All progress status in the application" should {
     "be mapped to the report labels" in {
       ProgressStatuses.allStatuses
-        .filterNot(_.key.contains("ASSESSMENT")) // TODO: Legacy statuses which should be removed
+        .filterNot(_.key.contains("ASSESSMENT")) // TODO: Legacy statuses which should be removed // btw, it's not legacy anymore
+          .filterNot(_.key.startsWith("FSB")) // TODO: Should FSB go to report ?
```
Comment (kjharkin) [inline it/model/report/ProgressStatusesMongoReportLabelsSpec.scala]: Can you remove TODOs please - if the requirement is unclear seek clarification

**Comment by kjharkin** [comment]
All good - just a couple of minor comments

File: it/model/report/ProgressStatusesMongoReportLabelsSpec.scala (hunk 2 above)
Comment (vglushak) [inline it/model/report/ProgressStatusesMongoReportLabelsSpec.scala]: @henricook already fixed this test in his branch, which is based on this branch. I would prefer to change less so he do not need to resolve conflicts.

File: app/repositories/application/GeneralApplicationRepository.scala (hunk 1 above)
Comment (vglushak) [inline app/repositories/application/GeneralApplicationRepository.scala]: ok.

File: it/model/report/ProgressStatusesMongoReportLabelsSpec.scala (hunk 2 above)
Comment (henricook) [inline it/model/report/ProgressStatusesMongoReportLabelsSpec.scala]: Yep i've removed todos and fixed this test

File: app/controllers/CandidateAllocationController.scala [hunk 3]
```diff
@@ -78,9 +78,14 @@ trait CandidateAllocationController extends BaseController {
     }
   }
 
-  def findCandidatesEligibleForEventAllocation(assessmentCenterLocation: String): Action[AnyContent] = Action.async {
+  def findCandidatesEligibleForEventAllocation(
+    assessmentCenterLocation: String,
```
Comment (henricook) [inline app/controllers/CandidateAllocationController.scala]: You might hate me for this, can this be `assessmentCentreLocation`? So we can consistently spell Centre the same throughout?

Comment (vglushak) [inline app/controllers/CandidateAllocationController.scala]: Ok.

**Pull request merged**